controller-server/src/main/java/com/yahoo/vespa/hosted/controller/LockedTenant.java

```diff
@@ -154,8 +155,11 @@ public CloudTenant get() {
 
         public Cloud withDeveloperKey(PublicKey key, Principal principal) {
             BiMap<PublicKey, Principal> keys = HashBiMap.create(developerKeys);
```

Comment:
Consider using `SimplePrincipal` in the map as well so that the equality contract is obvious.